### PR TITLE
Assorted buffer size fixes

### PR DESCRIPTION
**Pull request by djlowther**
During my current efforts to port MicroPython to Morello, I came across a handful of places where allocated buffers are too small for the operations that are performed on them. As they are only off by a few bytes, they likely have gone unnoticed, but they result in capability bounds faults on my Morello port.

Regardless, in the interest of memory safety, I've committed these fixes and packaged them into this pull request.

### CODE REVIEW

File: lib/oofatfs/ff.c [hunk 1]
```diff
@@ -2821,7 +2821,7 @@ static FRESULT create_name (    /* FR_OK: successful, FR_INVALID_NAME: could not
         if (di >= FF_MAX_LFN) return FR_INVALID_NAME;   /* Reject too long name */
         lfn[di++] = wc;                 /* Store the Unicode character */
     }
-    while (*p == '/' || *p == '\\') p++;    /* Skip duplicated separators if exist */
+    if(wc == '/' || wc == '\\') while (*p == '/' || *p == '\\') p++;    /* Skip duplicated separators if exist */
```
Comment (dpgeorge) [inline lib/oofatfs/ff.c:2824]: Can you please put a single space between the `if` and the `(`.

Also, please reword the title of the commit message for this commit to:
```
lib/oofatfs: Fix speculative read in create_name.
```

**Comment by dpgeorge** [comment]
Thank you for these fixes, they are definitely needed!

File: lib/oofatfs/ff.c (hunk 1 above)
Comment (djlowther) [reply lib/oofatfs/ff.c]: All fixed and pushed.

**Comment by dpgeorge** [comment]
Thanks for updating.

**Pull request merged**